plugins/woocommerce/src/Internal/Orders/OrderAttributionController.php

```diff
@@ -0,0 +1,428 @@
+		 *
+		 * @since x.x.x
+		 *
+		 * @param int $lifetime The lifetime of the cookie in months.
+		 */
+		$lifetime = (int) apply_filters( 'wc_order_attribution_cookie_lifetime_months', 6 );
+
+		/**
+		 * Filter the session length for source tracking.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param int $session_length The session length in minutes.
+		 */
+		$session_length = (int) apply_filters( 'wc_order_attribution_session_length_minutes', 30 );
+
+		/**
+		 * Filter to allow tracking.
+		 *
+		 * @since x.x.x
+		 *
+		 * @param bool $allow_tracking True to allow tracking, false to disable.
+		 */
+		$allow_tracking = wc_bool_to_string( apply_filters( 'wc_order_attribution_allow_tracking', true ) );
+
+		// Create Order Attribution JS namespace with parameters.
+		$namespace = array(
+			'params' => array(
+				'lifetime'      => $lifetime,
+				'session'       => $session_length,
+				'ajaxurl'       => admin_url( 'admin-ajax.php' ),
+				'prefix'        => $this->field_prefix,
+				'allowTracking' => $allow_tracking,
+			),
+		);
+
+		wp_localize_script( 'wc-order-attribution', 'wc_order_attribution', $namespace );
+	}
+
+	/**
+	 * Enqueue the stylesheet for admin pages.
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	private function enqueue_admin_scripts_and_styles() {
+		$screen = get_current_screen();
+		if ( $screen->id !== $this->get_order_screen_id() ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NotInFooter
+		wp_enqueue_script(
+			'woocommerce-order-attribution-admin-js',
+			plugins_url( "assets/js/admin/order-attribution-admin{$this->get_script_suffix()}.js", WC_PLUGIN_FILE ),
+			array( 'jquery' ),
+			Constants::get_constant( 'WC_VERSION' )
+		);
+	}
+
+	/**
+	 * Display the origin column in the orders table.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int $order_id The order ID.
+	 *
+	 * @return void
+	 */
+	private function display_origin_column( $order_id ): void {
+		try {
+			// Ensure we've got a valid order.
+			$order = $this->get_hpos_order_object( $order_id );
+			$this->output_origin_column( $order );
+		} catch ( Exception $e ) {
+			return;
+		}
+	}
+
+	/**
+	 * Output the data for the Origin column in the orders table.
+	 *
+	 * @param WC_Order $order The order object.
+	 *
+	 * @return void
+	 */
+	private function output_origin_column( WC_Order $order ) {
+		$source_type = $order->get_meta( $this->get_meta_prefixed_field( 'type' ) );
+		$source      = $order->get_meta( $this->get_meta_prefixed_field( 'utm_source' ) );
+		if ( ! $source ) {
+			$source = __( '(none)', 'woocommerce' );
+		}
+		echo esc_html( $this->get_origin_label( $source_type, $source ) );
+	}
+
+	/**
+	 * Add attribution hidden input fields for checkout & customer register froms.
+	 */
+	private function source_form_fields() {
+		foreach ( $this->fields as $field ) {
+			printf( '<input type="hidden" name="%s" value="" />', esc_attr( $this->get_prefixed_field( $field ) ) );
+		}
+	}
```

Comment:
As those are functionally incomplete without data fed by client-side JS, maybe we can move this logic - appending input elements to the form, to JS file (https://github.com/woocommerce/woocommerce/pull/39701/files#diff-21bb08f041552efddbc9e2515315dddcedadedef055476d4fe4dd35962d5c25fR57-R64) . To have the responsibility gathered in one place?
so PHP will just enqueue JS, which will fetch data from Sourcebuster, and add it to the form.

those elements are non-visual, so there is no benefit in server-side rendering them.

Reply:
This is a good idea as well, I think this can be done in a follow-up PR, hopefully before release.